vault/namespaces_store_test.go

```diff
@@ -88,8 +89,12 @@ func TestNamespaceStore(t *testing.T) {
 	require.Equal(t, ns[0].UUID, itemUUID)
 
 	// Delete that item.
-	err = s.DeleteNamespace(ctx, itemUUID)
+	status, err := s.DeleteNamespace(ctx, itemUUID)
 	require.NoError(t, err)
+	require.Equal(t, "in-progress", status)
+
+	// wait until deletion
+	time.Sleep(10 * time.Millisecond)
```

Comment:
I suspect this delay should be longer for CI:

```

2025-04-24T20:38:03.0377776Z 2025-04-24T20:38:02.868Z [DEBUG] TestNamespaceStore.core: persisting feature flags
2025-04-24T20:38:03.0377955Z     namespaces_store_test.go:102: 
2025-04-24T20:38:03.0378407Z         	Error Trace:	/home/runner/work/openbao/openbao/vault/namespaces_store_test.go:102
2025-04-24T20:38:03.0378703Z         	Error:      	Should be empty, but was [ID: Xt4pVL. Path: ns1/]
2025-04-24T20:38:03.0378902Z         	Test:       	TestNamespaceStore
2025-04-24T20:38:03.0379098Z 2025-04-24T20:38:02.967Z [DEBUG] TestNamespaceStore.core: shutdown called
```


or better, use the retry logic in `sys_namespaces_test.go`